### PR TITLE
update CI for publishing to Upbound registry

### PR DESCRIPTION
**Pull request by jastang**
### Description of your changes

First step to publish to `notone/provider-ksyun` on the Upbound registry.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

N/A; ensure local build functional:

```
10:04:46 [ OK ] docker build build-6c180cc5/provider-ksyun-arm64
10:04:46 [ .. ] Building package provider-ksyun-v0.0.0-3.gc9d79c4.dirty.xpkg for linux_arm64
xpkg saved to /Users/jasontang/workspace/provider-ksyun/_output/xpkg/linux_arm64/provider-ksyun-v0.0.0-3.gc9d79c4.dirty.xpkg
10:04:51 [ OK ] Built package provider-ksyun-v0.0.0-3.gc9d79c4.dirty.xpkg for linux_arm64
```

### CODE REVIEW

File: package/crossplane.yaml [hunk 1]
```diff
@@ -2,3 +2,10 @@ apiVersion: meta.pkg.crossplane.io/v1
 kind: Provider
 metadata:
   name: provider-ksyun
+  annotations:
+    meta.crossplane.io/maintainer: <ADD EMAIL>
+    meta.crossplane.io/source: github.com/notone0010/provider-ksyun
+    meta.crossplane.io/description: |
+      A Crossplane provider that exposes managed resources for the Ksyun API.
+    meta.crossplane.io/readme: |
+      <ADD README>
```
Comment (jastang) [inline package/crossplane.yaml:11]: @notone0010 feel free to change the metadata here, which will render in the Upbound listing. all object annotations described in the [xpkg spec](https://github.com/crossplane/crossplane/blob/master/contributing/specifications/xpkg.md#object-annotations) will be honoured.

Comment (notone0010) [reply package/crossplane.yaml:11]: @jastang I have merged and completed the readme.

Comment (notone0010) [reply package/crossplane.yaml:11]: And next, what need I do for the package publish?

Comment (jastang) [reply package/crossplane.yaml:11]: @notone0010 you'll need to add Github Actions Secrets to this repository. See the keys in ci.yml here:

https://github.com/notone0010/provider-ksyun/actions/runs/6610514274/workflow#L264-L265

That can be either be a [robot](https://docs.upbound.io/concepts/accounts/robots/) token/id pair, or your own personal credentials.

Comment (notone0010) [reply package/crossplane.yaml:11]: > @notone0010 you'll need to add Github Actions Secrets to this repository. See the keys in ci.yml here:
> 
> https://github.com/notone0010/provider-ksyun/actions/runs/6610514274/workflow#L264-L265
> 
> That can be either be a [robot](https://docs.upbound.io/concepts/accounts/robots/) token/id pair, or your own personal credentials.

@jastang I still encountered an authentication error when I set Github Actions Secrets. 

The Github Actions Secrets settings' screenshot
image

Error details
image

Comment (jastang) [reply package/crossplane.yaml:11]: This specific workflow is not well-documented at the moment, but the recommended path is to use a robot token. 

You'll need to assign your robot to a team, and give that team WRITE permissions to `noteone/provider-ksyun` repo. See https://docs.upbound.io/concepts/accounts/teams/#manage-repository-permissions.

If we're still running into issues, let's open an issue to track this so we don't get lost in the merged PR 👍🏼

**Pull request merged**